spec/system/admin/tos_banner_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+describe 'Terms of Service banner' do
+  include AuthenticationHelper
+
+  let(:admin_user) { create(:admin_user, terms_of_service_accepted_at: nil) }
+
+  before do
+    Spree::Config.enterprises_require_tos = true
+    login_as admin_user
+  end
+
+  context "when not accepted" do
+    it "shows banner" do
+      visit '/admin'
+
+      expect(page).to have_content("Terms of Service have been updated")
+
+      # Click on the accept button
+      expect do
+        click_button "Accept Terms of Service"
+        admin_user.reload
+      end.to change { admin_user.terms_of_service_accepted_at }
+      expect(page).to_not have_content("Terms of Service have been updated")
+
+      # Check the banner doesn't show again once ToS has been accepted
+      page.refresh
+      expect(page).to_not have_content("Terms of Service have been updated")
+    end
+  end
+
+  context "when updating Terms of Service" do
+    let(:test_file_path) { "public/Terms-of-service.pdf" }
+
+    it "shows the banner" do
+      # ToS has been accepted
+      admin_user.update!(terms_of_service_accepted_at: 2.days.ago)
+
+      # Upload new ToS
+      visit admin_terms_of_service_files_path
+      attach_file "Attachment", Rails.root.join(test_file_path)
+      click_button "Create Terms of service file"
+
+      # check it has been uploaded
+      expect(page).to have_link "Terms of Service"
+
+      expect(page).to have_content("Terms of Service have been updated")
+    end
+  end
```

Comment:
I would suggest that we just need to test showing/hiding, and don't need this second spec about exact conditions of when it shows/hides, because that's covered in the request spec. But maybe that's just personal preference.

Reply:
However, I would at least suggest that instead simulating the upload of new ToS file, you could just update it directly in the model, which would be a lot quicker.
Uploading of the new ToS is covered in `spec/system/admin/configuration/terms_of_service_files_spec.rb`.

Reply:
I thought that, too, and just went to do this, *but* this test has only one page visit and a form submission. The test conditions are then carried out on the same page. Replacing this doesn't result in any speed-up of the test. So it's better to run real user interaction instead of manual database changes then.